From 4e0729bd86a6a702322071604450bacdd8a60355 Mon Sep 17 00:00:00 2001
From: Nikita Ofitserov <himikof@gmail.com>
Date: Tue, 23 Aug 2022 20:27:39 +0300
Subject: [PATCH] Fix Metalink4 parsing with foreign namespaces

Rename local checkNsUri helper function to make boolean conditions easily readable,
as the issue was not apparent with the old helper function name.
---
 src/MetalinkParserStateV4Impl.cc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/MetalinkParserStateV4Impl.cc b/src/MetalinkParserStateV4Impl.cc
index 43fdeec..4fee5c3 100644
--- a/src/MetalinkParserStateV4Impl.cc
+++ b/src/MetalinkParserStateV4Impl.cc
@@ -47,7 +47,7 @@ namespace aria2 {
 const char METALINK4_NAMESPACE_URI[] = "urn:ietf:params:xml:ns:metalink";
 
 namespace {
-bool checkNsUri(const char* nsUri)
+bool isMetalink4Ns(const char* nsUri)
 {
   return nsUri && strcmp(nsUri, METALINK4_NAMESPACE_URI) == 0;
 }
@@ -57,7 +57,7 @@ void MetalinkMetalinkParserStateV4::beginElement(
     MetalinkParserStateMachine* psm, const char* localname, const char* prefix,
     const char* nsUri, const std::vector<XmlAttr>& attrs)
 {
-  if (checkNsUri(nsUri) && strcmp(localname, "file") != 0) {
+  if (!isMetalink4Ns(nsUri) || strcmp(localname, "file") != 0) {
     psm->setSkipTagState();
     return;
   }
@@ -83,7 +83,7 @@ void FileMetalinkParserStateV4::beginElement(MetalinkParserStateMachine* psm,
                                              const char* nsUri,
                                              const std::vector<XmlAttr>& attrs)
 {
-  if (!checkNsUri(nsUri)) {
+  if (!isMetalink4Ns(nsUri)) {
     psm->setSkipTagState();
   }
   else if (strcmp(localname, "size") == 0) {
@@ -293,7 +293,7 @@ void PiecesMetalinkParserStateV4::beginElement(
     MetalinkParserStateMachine* psm, const char* localname, const char* prefix,
     const char* nsUri, const std::vector<XmlAttr>& attrs)
 {
-  if (checkNsUri(nsUri) && strcmp(localname, "hash") == 0) {
+  if (isMetalink4Ns(nsUri) && strcmp(localname, "hash") == 0) {
     psm->setPieceHashStateV4();
   }
   else {
-- 
2.25.1

